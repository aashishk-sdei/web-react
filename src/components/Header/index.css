.header {
  @apply fixed top-0 left-0 w-full z-2000;
}

.main-wrapper {
  @apply px-3 md:px-6 relative;
}
.header .main-wrapper {
  @apply px-3
}
.header .header-section {
  @apply flex md:items-center md:py-2 md:flex-row;
  min-height: min-content;
}
.main-nav {
  @apply block md:flex space-x-4 md:mr-auto ml-6;
}
.main-nav a {
  @apply mx-0 !important md:mx-1 my-4 md:my-0 text-sm leading-3 font-medium text-gray-5 px-3 py-1 rounded-md min-h-8 flex items-center relative
  hover:text-gray-6 hover:bg-gray-2;
  font-family: "Atlas Grotesk Web Medium", sans-serif !important;
}
a:focus,
button:focus
 {
  outline-color: #295DA1;
}
.main-nav a.active {
  @apply bg-transparent hover:bg-transparent text-gray-7 !important;
}
.main-nav a.active::after,
.nav-link.notifications-link.active:after {
  content: " ";
  @apply h-1 w-full absolute -bottom-3 left-0 bg-red;
}
.nav-link.clues-link.active:after {
  content: " ";
  @apply h-1 w-full absolute -bottom-3 left-0 bg-red;
}

.avatar-dropdown {
  @apply cursor-pointer hidden md:block border-2 border-white rounded-full p-0.5;
}
.avatar-dropdown:hover {
  @apply border-2 border-gray-2 rounded-full p-0.5;
}
.avatar-dropdown:focus {
  @apply border-2 border-skyblue-2 rounded-full p-0.5;
}
.avatar-dropdown div:nth-child(2) {
  top: 34px !important;
  right: 0;
  transform: initial !important;
  left: initial !important;
}
.hamburger-menu {
  @apply absolute right-2 cursor-pointer z-1000 block md:hidden;
}
.main-nav-mobile {
  @apply hidden;
}
.my-account {
  @apply block md:hidden ml-0 !important;
}

.nav-link.notifications-link {
@apply bg-gray-2 w-8 h-8 flex items-center justify-center absolute md:relative mr-4 rounded-full right-12 md:right-auto md:top-auto z-10;
}

.nav-link.notifications-link.active svg path {
stroke: #FC4040 !important;
}
.nav-link.notifications-link .ntf-count {
  min-width: 16px;
  left:24px;
  padding:0 3px;
}
.nav-link.clues-link {
  @apply bg-gray-2 w-8 h-8 flex items-center justify-center absolute md:relative rounded-full right-12 md:right-auto md:top-auto z-10;
  margin-right: 1.3rem;
}
.nav-link.clues-link.active svg path {
  stroke: none !important;
}
.nav-link.clues-link .ntf-count {
  min-width: 16px;
  left:24px;
  padding:0 3px;
}
.mb-nav {
  @apply mt-1 pt-1 mb-1;
  margin-left: -0.75em !important;
}
.mb-nav a {
  @apply text-gray-5;
}

@media screen and (max-width: 767px) {
  .nav-link.clues-link {
    right: 6rem;
    top: 9px;
  }
  .header .header-section {
  @apply flex  items-start py-3 flex-col;
  min-height: min-content
  }
  .main-nav {
    @apply hidden;
  }
  .main-nav-mobile {
    @apply flex space-x-4 flex-col w-full mt-6 z-1000;
  }
  .main-nav-mobile a {
    @apply ml-0 !important px-3 py-2 block;
  }
  .main-nav-mobile .people-dd {
    margin-left: 0 !important;
  }
  .nav-link.notifications-link.active {
    @apply bg-gray-2
  } 
  .nav-link.notifications-link.active:after {
    display: none;
  }  
  .nav-link.notifications-link {
    top: 9px;
  }
  .nav-link.clues-link.active {
    @apply bg-gray-2
  }
  .nav-link.clues-link.active:after {
    display: none;
  }
}
