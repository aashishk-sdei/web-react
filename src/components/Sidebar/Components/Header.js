const Header = ({
    profile
}) => {
    return <div className="px-4 sm:px-3.5 border-b border-gray-6 mt-1 pb-3.5">
        {profile.name ? <h2 id="slide-over-heading" className="text-2xl font-bold text-white mb-0 pb-0">
            <span>
                {profile.name && profile.name.value && profile.name.value.firstName}
            </span>
            <span className="inline-block ml-1">
                {profile.name && profile.name.value && profile.name.value.lastName}
            </span>
        </h2> : null}
        <p className="font-light text-sm mt-0.5 text-gray-3 mb-0">
            {profile.namePlace || ""}
        </p>
    </div>
}
export default Header;