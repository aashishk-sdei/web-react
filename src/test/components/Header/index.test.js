import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from "react-redux";
import store from "../../../redux/store";
import Header from '../../../components/Header';

test('should test Header component with list of props', () => {
    const wrapper = shallow(
        <Provider store={store}>
            <Header />
        </Provider>
    );
    expect(wrapper).toMatchSnapshot();
});