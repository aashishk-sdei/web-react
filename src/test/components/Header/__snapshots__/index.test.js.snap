// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Header component with list of props 1`] = `ShallowWrapper {}`;
